extensions/web/09_file.js

```diff
@@ -67,58 +69,57 @@
     return result;
   }
 
-  /**
-   * @param  {...Uint8Array} bytesArrays
-   * @returns {Uint8Array}
-   */
-  function concatUint8Arrays(...bytesArrays) {
-    let byteLength = 0;
-    for (const bytes of bytesArrays) {
-      byteLength += bytes.byteLength;
-    }
-    const finalBytes = new Uint8Array(byteLength);
-    let current = 0;
-    for (const bytes of bytesArrays) {
-      finalBytes.set(bytes, current);
-      current += bytes.byteLength;
+  /** @param {(Blob | Uint8Array)[]} parts */
+  async function * toIterator (parts) {
+    for (const part of parts) {
+      if (part instanceof Blob) {
+        yield * part.stream();
+      } else if (ArrayBuffer.isView(part)) {
+        let position = part.byteOffset;
+        const end = part.byteOffset + part.byteLength;
+        while (position !== end) {
+          const size = Math.min(end - position, POOL_SIZE);
+          const chunk = part.buffer.slice(position, position + size);
+          position += chunk.byteLength;
+          yield new Uint8Array(chunk);
+        }
+      }
     }
-    return finalBytes;
   }
 
   /** @typedef {BufferSource | Blob | string} BlobPart */
 
   /**
      * @param {BlobPart[]} parts
      * @param {string} endings
-     * @returns {Uint8Array}
      */
   function processBlobParts(parts, endings) {
-    /** @type {Uint8Array[]} */
+    /** @type {(Uint8Array|Blob)[]} */
     const bytesArrays = [];
```

Comment:
Maybe `parts` would be a better name here? Also could you update the `@returns` to be correct?